table/block_based_table_reader.cc

```diff
@@ -129,18 +129,17 @@ void ForceReleaseCachedEntry(void* arg, void* h) {
   cache->Release(handle, true /* force_erase */);
 }
 
-Cache::Handle* GetEntryFromCache(Cache* block_cache, const Slice& key,
-                                 int level, Tickers block_cache_miss_ticker,
-                                 Tickers block_cache_hit_ticker,
-                                 uint64_t* block_cache_miss_stats,
-                                 uint64_t* block_cache_hit_stats,
-                                 Statistics* statistics,
-                                 GetContext* get_context) {
+Cache::Handle* GetEntryFromCache(
+    Cache* block_cache, const Slice& key, const BlockBasedTable::Rep* rep,
+    Tickers block_cache_miss_ticker, Tickers block_cache_hit_ticker,
+    uint64_t* block_cache_miss_stats, uint64_t* block_cache_hit_stats,
+    Statistics* statistics, GetContext* get_context) {
+  assert(rep);
```

Comment:
If we even need to pass `rep` to this function, is there any reason not to make it a function of class `BlockBasedTable`?

Reply:
No. The class has some static functions. I'm doing a major refactoring to remove them right now. I will send out a new PR for it.

Reply:
Thanks for your comment. I refactored the code to make GetEntryFromCache to become a member function of BlockBasedTable.